app/lib/actions/pulp3/capsule_content/create_publication.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Pulp3
+    module CapsuleContent
+      class CreatePublication < Pulp3::AbstractAsyncTask
+        middleware.use Actions::Middleware::ExecuteIfContentsChanged
+        def plan(repository, smart_proxy, options)
+          sequence do
+            if !::Katello::RepositoryTypeManager.find(repository.content_type).pulp3_skip_publication
+              action = plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id, :contents_changed => options[:contents_changed], :options => options)
+              plan_action(RefreshDistribution, repository, smart_proxy,
+                            :tasks => action.output,
+                            :use_repository_version => false,
+                            :contents_changed => options[:contents_changed])
+            else
+              plan_action(RefreshDistribution, repository, smart_proxy,
+                            :tasks => action.output,
```

Comment:
action here is undefined, so it won't work properly :)  should refresh_distribution look up the latest version?

Reply:
oh it looks like refreshDistribution doesn't need an action if use_repository_version is true, so maybe just omit this